core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,307 @@
+          }
+        }
+        return defaultRecord;
+      default:
+        throw new IllegalArgumentException(String.format("Type: %s is not supported", type));
+    }
+  }
+
+  public static Object fromJson(Type type, String defaultValue) {
+    try {
+      JsonNode defaultValueJN = JsonUtil.mapper().readTree(defaultValue);
+      return fromJson(type, defaultValueJN);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Failed to parse: " + defaultValue + "; reason: " + e.getMessage(), e);
+    }
+  }
+
+  public static String toJson(Type type, Object defaultValue) {
+    return toJson(type, defaultValue, false);
+  }
+
+  public static String toJson(Type type, Object defaultValue, boolean pretty) {
+    try {
+      StringWriter writer = new StringWriter();
+      JsonGenerator generator = JsonUtil.factory().createGenerator(writer);
+      if (pretty) {
+        generator.useDefaultPrettyPrinter();
+      }
+      toJson(type, defaultValue, generator);
+      generator.flush();
+      return writer.toString();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static void toJson(Type type, Object javaDefaultValue, JsonGenerator generator)
+      throws IOException {
+    switch (type.typeId()) {
+      case BOOLEAN:
+        generator.writeBoolean((boolean) javaDefaultValue);
+        break;
+      case INTEGER:
+        generator.writeNumber((int) javaDefaultValue);
+        break;
+      case LONG:
+        generator.writeNumber((long) javaDefaultValue);
+        break;
+      case FLOAT:
+        generator.writeNumber((float) javaDefaultValue);
+        break;
+      case DOUBLE:
+        generator.writeNumber((double) javaDefaultValue);
+        break;
+      case DATE:
+        generator.writeString(DateTimeUtil.formatEpochDays((int) javaDefaultValue));
+        break;
+      case TIME:
+        generator.writeString(DateTimeUtil.formatTimeOfDayMicros((long) javaDefaultValue));
+        break;
+      case TIMESTAMP:
+        generator.writeString(DateTimeUtil.formatEpochTimeMicros(
+            (long) javaDefaultValue,
+            ((Types.TimestampType) type).shouldAdjustToUTC()));
+        break;
+      case STRING:
+        generator.writeString((String) javaDefaultValue);
+        break;
+      case UUID:
+        generator.writeString(javaDefaultValue.toString());
+        break;
+      case FIXED:
+      case BINARY:
+        generator.writeString(BaseEncoding.base16().encode(ByteBuffers.toByteArray(((ByteBuffer) javaDefaultValue))));
+        break;
+      case DECIMAL:
+        generator.writeNumber((BigDecimal) javaDefaultValue);
+        break;
+      case LIST:
+        List<Object> defaultList = (List<Object>) javaDefaultValue;
+        Type elementType = type.asListType().elementType();
+        generator.writeStartArray();
+        for (Object elementDefault : defaultList) {
```

Comment:
Minor: this isn't the element default, it is one element in the default list. I think this should just be `element`.